### PR TITLE
add bors

### PR DESCRIPTION
**Pull request by japaric**

### CODE REVIEW

**Comment by japaric** [comment]
bors ping

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,3 +38,25 @@ jobs:
...
+
+  ci-failure:
+    name: ci
+    if: ${{ !success() }}
+    needs:
+      - test
+    runs-on: ubuntu-20.04
+    steps:
+      - name: CI failed
+        run: exit 1
```
Comment (jonas-schievink) [inline .github/workflows/ci.yml:62]: Hmm, why are these needed? Can't you add all required workflows to bors.toml instead? `cortex-m` doesn't have this, for example.

File: bors.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+delete_merged_branches = true
+status = ["ci"]
```
Comment (jonas-schievink) [inline bors.toml:2]: This file can be moved to `.github/bors.toml` to keep the root dir clean

File: .github/workflows/ci.yml (hunk 1 above)
Comment (japaric) [reply .github/workflows/ci.yml:62]: unsure; copied them from a working GHA+bors configuration (heapless)
